CodeDrivers/CodeDrivers/CodeDrivers/Repository/CarRepository.cs

```diff
@@ -68,15 +69,21 @@ public List<Car> GetAllAvailable()
             return availableCars;
         }
 
-        public void DisplayAllItems(List<Car> cars)
+        public void DisplayItems(List<Car> cars)
         {
             
-            Console.WriteLine("Wszystkie samochody z naszej ofery: ");
+            //.WriteLine("Wszystkie samochody z naszej ofery: ");
             foreach (var item in cars)
             {
                 Console.WriteLine($"{item.Id.ToString()}. {item.Brand.ToString()}, {item.Model}, {item.Segment}, {item.GearTransmission}, {item.IsAvailable}");
             }
         }
+
+        public List<Car> GetAllExceptCarsWithProvidedIds(List<int> providedCarIds)
```

Comment:
dokładnie tego użyj "poziom wyżej" jeśli chcesz zachować generyczność. Ewentualnie możesz stworzyć ICarRepository dziedziczące po IRepository i ono może zawierać definicję metody GetAllExceptCarsWithProvidedIds

Reply:
chyba, że IRepository ma być tylko dla Car, to wtedy bez sensu jest <T>